### PR TITLE
GH-37337: [MATLAB] Add `arrow.array.Time64Array` class

### LINKED ISSUE #37337: [MATLAB] Add `arrow.array.Time64Array` class
### Describe the enhancement requested

Now that `arrow.type.Time64Type` class has been added to the MATLAB Interface (#37287), we can add the `arrow.array.Time64Array` class.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

Now that `arrow.type.Time64Type` class has been added to the MATLAB Interface (#37287), we can add the `arrow.array.Time64Array` class.

### What changes are included in this PR?

1. New `arrow.array.Time64Array` class.
2. New `arrow.type.traits.Time64Traits` class.
3. Enhanced `arrow.array(data)` to return a `arrow.array.Time64Array` instance if the input `data` is a MATLAB `duration` array. The MATLAB `duration` type represents length of time in fixed-length units.
4. Enhanced `arrow.type.traits.traits(id)` to return an `arrow.type.traits.Time64Traits` instance if `id` is an `arrow.type.ID.Time64` value or equal to the string `"duration".`
5. Updated the test utility `arrow.internal.test.tabular.createAllSupportedArrayTypes()` to include an `arrow.array.Time64Array` instance in its output.
6. Updated the test utility `arrow.internal.test.tabular.createTableWithSupportedTypes()` to include a `duration` variable on the returned `table`. 
7. Updated the README.md to include `Time32Array` and `Time64Array`.

**Example Usage**
```matlab

% Create a duration array
>> ms = milliseconds([100 150 1001])

ms = 

  1×3 duration array

     0.1 sec    0.15 sec   1.001 sec

% Create an arrow.type.Time64Array
>> time = arrow.array(ms)

time = 

[
  00:00:00.100000,
  00:00:00.150000,
  00:00:01.001000
]

% Create a duration array from an arrow.array.Time64Array
>> data = duration(time) 

data = 

  3×1 duration array

     0.1 sec
    0.15 sec
   1.001 sec

```

### Are these changes tested?

Yes.

1. Added a new test class `tTime64Array.m`.
2. Added a new test class `tTime64Traits.m`
3. Added new test cases to `tArray.m` and `ttraits.m`

### Are there any user-facing changes?

Yes. Users can now create `arrow.array.Time64Array`s from  MATLAB `duration`s

* Closes: #37337

### CODE REVIEW

File: matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc
Comment (kou) [inline matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc:1]: It seems that most codes are same as `time64_array.cc`.
Can we unify them?

File: matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+       const mda::TypedArray<mda::MATLABString> units_mda = opts[0]["TimeUnit"];
+
+       // extract the time unit
+       const std::u16string& u16_timeunit = units_mda[0];
+       MATLAB_ASSIGN_OR_ERROR(const auto time_unit,
+                              arrow::matlab::type::timeUnitFromString(u16_timeunit),
+                              error::UKNOWN_TIME_UNIT_ERROR_ID)
+
+       // create the Time64Type
+       auto data_type = arrow::time64(time_unit);
```
Comment (kou) [inline matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc:48]: Note that valid `time_unit` for `arrow::time64()` is only `arrow::TimeUnit::MICRO` and `arrow::TimeUnit::NANO`.
We may want to validate `time_unit`.

File: matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc
Comment (sgilmore10) [reply matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc:1]: That's a good point. I'll add a function templated called `make_time_array` that both `Time64Array` and `Time32Array` can use.

File: matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc (hunk 1 above)
Comment (sgilmore10) [reply matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc:48]: Will do! I'll add a `validate_time_unit` function.

Comment (sgilmore10) [reply matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc]: I added a new function called `validateTimeUnit`.

File: matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc
Comment (sgilmore10) [reply matlab/src/cpp/arrow/matlab/array/proxy/time64_array.cc:1]: I added two new function templates: `make_time_array` and `make_time_type`. The former is called by both `Time64Array`'s and `Time32Array`'s `make` function. The latter is called by both `Time64Type`'s and `Time32Type`'s `make` function. Doing so removed duplicate code across these classes.

File: matlab/test/tfeather.m [hunk 2]
```diff
@@ -282,3 +269,9 @@ function UnicodeVariableNames(testCase)
 
     end
 end
+
+function t = createFeatherCompatibleTable()
+    import arrow.internal.test.tabular.createTableWithSupportedTypes
+    t = createTableWithSupportedTypes;
+    t = removevars(t, "duration");
```
Comment (kevingurney) [inline matlab/test/tfeather.m:276]: Could you add a comment here saying that Feather V1 does not support the `Time[x][y]` types?

File: matlab/src/cpp/arrow/matlab/array/proxy/time_array.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+       using MatlabBuffer = arrow::matlab::buffer::MatlabBuffer;
+       using TimeArray =  arrow::NumericArray<ArrowType>; 
+       using TimeArrayProxy = proxy::NumericArray<ArrowType>;
+       using CType = typename arrow::TypeTraits<ArrowType>::CType;
+
+       mda::StructArray opts = constructor_arguments[0];
+
+       const mda::TypedArray<CType> time_mda = opts[0]["MatlabArray"];
+       const mda::TypedArray<bool> validity_bitmap_mda = opts[0]["Valid"];
+       const mda::TypedArray<mda::MATLABString> units_mda = opts[0]["TimeUnit"];
```
Comment (kevingurney) [inline matlab/src/cpp/arrow/matlab/array/proxy/time_array.h:45]: `units_mda` -> `time_unit_mda`

File: matlab/src/cpp/arrow/matlab/array/proxy/time_array.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+       // extract the time unit
+       const std::u16string& u16_timeunit = units_mda[0];
```
Comment (kevingurney) [inline matlab/src/cpp/arrow/matlab/array/proxy/time_array.h:48]: `time_unit_utf16`

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thanks for working on this! Looks really good!

**Review by kevingurney — APPROVED** [review]

**Comment by kevingurney** [comment]
+1

**Pull request merged**